### PR TITLE
feat(aws,comprehend): initial connector code

### PR DESCRIPTION
**Pull request by igpetrov**
feat(aws,comprehend): initial connector code
[comprehend documentation](https://github.com/camunda/camunda-docs/pull/4391)
[issue](https://github.com/camunda/connectors/issues/2924)

### CODE REVIEW

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+
+    if (operationType.equals("sync")) {
+      return mapper.convertValue(node, ComprehendSyncRequestData.class);
+    }
+    if (operationType.equals("async")) {
+      return mapper.convertValue(node, ComprehendAsyncRequestData.class);
+    }
+    throw new UnsupportedOperationException("Unsupported execution type: " + operationType);
+  }
```
Comment (sbuettner) [inline connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java:34]: This should work by just using jackson annotations. You can use a default mapping in jackson.

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  private String determineActionType(JsonNode node) {
+    if (node.has("type")) {
+      return node.get("type").asText();
+    }
+    throw new UnsupportedOperationException("Unsupported operation type");
+  }
```
Comment (sbuettner) [inline connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java:41]: Same here

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java (hunk 1 above)
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java (hunk 2 above)
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/AsyncComprehendCaller.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.Map;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AsyncComprehendCaller
+    implements ComprehendCaller<StartDocumentClassificationJobResult, ComprehendAsyncRequestData> {
+
+  public static final String VPC_CONFIG_EX =
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/AsyncComprehendCaller.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    documentationRef =
+        "https://docs.camunda.io/docs/next/components/connectors/out-of-the-box-connectors/amazon-comprehend/",
+    icon = "icon.svg")
+public class ComprehendConnectorFunction implements OutboundConnectorFunction {
+
+  private ComprehendClientSupplier clientSupplier;
+
+  private SyncComprehendCaller syncComprehendCaller;
+
+  private AsyncComprehendCaller asyncComprehendCaller;
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendExecutionType.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ *       under one or more contributor license agreements. Licensed under a proprietary license.
+ *       See the License.txt file for more information. You may not use this file
+ *       except in compliance with the proprietary license.
+ */
+package io.camunda.connector.comprehend.model;
+
+public enum ComprehendExecutionType {
+  SYNC,
+  ASYNC
+}
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendExecutionType.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  public Object execute(OutboundConnectorContext context) throws Exception {
+    var request = context.bindVariables(ComprehendRequest.class);
+    ComprehendRequestData requestData = request.getInput();
+    if (requestData instanceof ComprehendSyncRequestData) {
+      return syncComprehendCaller.call(
+          clientSupplier.getSyncClient(request), (ComprehendSyncRequestData) requestData);
+    }
+    return asyncComprehendCaller.call(
+        clientSupplier.getAsyncClient(request), (ComprehendAsyncRequestData) requestData);
+  }
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java]: partially changed ))

File: connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import io.camunda.connector.comprehend.supplier.ComprehendClientSupplier;
+import io.camunda.connector.test.outbound.OutboundConnectorContextBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ComprehendConnectorFunctionTest {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java:32]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      ComprehendRequestData requestData) {
+    if (requestData.getDocumentReadAction().equals(ComprehendDocumentReadAction.NO_DATA)) {
+      return Optional.empty();
+    }
+
+    var documentReaderConfig =
+        new DocumentReaderConfig()
+            .withDocumentReadAction(requestData.getDocumentReadAction().name());
+
+    if (!requestData.getDocumentReadMode().equals(ComprehendDocumentReadMode.NO_DATA)) {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        throw new IllegalArgumentException(READ_ACTION_WITHOUT_FEATURES_EX);
+      }
+      documentReaderConfig.withFeatureTypes(features);
+    }
+
+    return Optional.of(documentReaderConfig);
+  }
+
+  default List<String> prepareFeatures(ComprehendRequestData requestData) {
+    List<String> features = new ArrayList<>(2);
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendDocumentReadAction.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+ *       under one or more contributor license agreements. Licensed under a proprietary license.
+ *       See the License.txt file for more information. You may not use this file
+ *       except in compliance with the proprietary license.
+ */
+package io.camunda.connector.comprehend.model;
+
+public enum ComprehendDocumentReadAction {
+  TEXTRACT_DETECT_DOCUMENT_TEXT,
+  TEXTRACT_ANALYZE_DOCUMENT,
+
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendDocumentReadAction.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+      ComprehendClientSupplier clientSupplier,
+      SyncComprehendCaller syncComprehendCaller,
+      AsyncComprehendCaller asyncComprehendCaller) {
+    this.clientSupplier = clientSupplier;
+    this.syncComprehendCaller = syncComprehendCaller;
+    this.asyncComprehendCaller = asyncComprehendCaller;
+  }
+
+  @Override
+  public Object execute(OutboundConnectorContext context) throws Exception {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/ComprehendConnectorFunction.java]: done

File: connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  @Mock private ComprehendClientSupplier clientSupplier;
+
+  @Mock private SyncComprehendCaller syncComprehendCaller;
+
+  @Mock private AsyncComprehendCaller asyncComprehendCaller;
+
+  @InjectMocks private ComprehendConnectorFunction comprehendConnectorFunction;
+
+  @Test
+  void executeSyncRequest() throws Exception {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java]: done

File: connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+    when(clientSupplier.getSyncClient(any())).thenCallRealMethod();
+    when(syncComprehendCaller.call(any(), any())).thenReturn(new ClassifyDocumentResult());
+
+    var result = comprehendConnectorFunction.execute(outBounderContext);
+    assertThat(result).isInstanceOf(ClassifyDocumentResult.class);
+  }
+
+  @Test
+  void executeAsyncRequest() throws Exception {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/test/java/io/camunda/connector/comprehend/ComprehendConnectorFunctionTest.java]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/AsyncComprehendCaller.java [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+      docClassificationRequest.withDocumentClassifierArn(asyncRequest.documentClassifierArn());
+    }
+
+    if (StringUtils.isNoneBlank(asyncRequest.flywheelArn())) {
+      docClassificationRequest.withFlywheelArn(asyncRequest.flywheelArn());
+    }
+
+    docClassificationRequest.withInputDataConfig(prepareInputConfig(asyncRequest));
+
+    if (StringUtils.isNoneEmpty(asyncRequest.jobName())) {
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/AsyncComprehendCaller.java]: done

File: connectors/aws/aws-comprehend/README.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# AWS Comprehend Connector
+
+TODO: add proper description
```
Comment (sbuettner) [inline connectors/aws/aws-comprehend/README.md:3]: @DenovVasil Can you provide a proper description? 😄

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+    if (requestData.getDocumentReadAction() == TEXTRACT_ANALYZE_DOCUMENT) {
+      List<String> features = prepareFeatures(requestData);
+      if (features.isEmpty()) {
+        LOGGER.warn("DocumentReadAction: TEXTRACT_ANALYZE_DOCUMENT, but features not selected.");
+        throw new IllegalArgumentException(READ_ACTION_WITHOUT_FEATURES_EX);
+      }
+      documentReaderConfig.withFeatureTypes(features);
+    }
+
+    return Optional.of(documentReaderConfig);
```
Comment (sbuettner) [inline connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java:61]: Why the need for optional here?

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendSyncRequestData.java [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+            label = "Text",
+            description = "The document text to be analyzed.")
+        @NotNull
+        String text,
+    @TemplateProperty(
+            id = "sync.documentReadAction",
+            label = "Document read action",
+            group = "input",
+            type = TemplateProperty.PropertyType.Dropdown,
+            defaultValue = "TEXTRACT_DETECT_DOCUMENT_TEXT",
```
Comment (sbuettner) [inline connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendSyncRequestData.java:29]: Is it "TEXTRACT_DETECT_DOCUMENT_TEXT" ? Or rather comprehend related?

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java (hunk 16 above)
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java:61]: this [DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax) is not mandatory.
it seems to me that the optional check looks pretty
`  new ClassifyDocumentRequest()
            .withText(requestData.text())
            .withEndpointArn(requestData.endpointArn())
            .withDocumentReaderConfig(prepareDocumentReaderConfig(requestData).orElse(null));`

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendSyncRequestData.java (hunk 17 above)
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendSyncRequestData.java:29]: part of comprehend https://docs.aws.amazon.com/comprehend/latest/APIReference/API_DocumentReaderConfig.html#comprehend-Type-DocumentReaderConfig-DocumentReadAction

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java (hunk 16 above)
Comment (sbuettner) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java:61]: if you can pass null, you could also return null right?

**Review by johnBgood — COMMENTED** [review]
Minor comments only 👍

File: connectors/aws/aws-comprehend/README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/README.md]: I am guilty

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+      "If you chose TEXTRACT_ANALYZE_DOCUMENT as the read action, "
+          + "you must specify one feature types";
+
+  Logger LOGGER = LoggerFactory.getLogger(ComprehendCaller.class);
+
+  int INITIAL_FEATURES_CAPACITY = 2;
+
+  T call(AmazonComprehendClient client, R requestData);
+
+  default Optional<DocumentReaderConfig> prepareDocumentReaderConfig(
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java:38]: Honestly, I'm not a fan of using `tryPrepare....`
But thank you very much for the offer.

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+            feel = Property.FeelMode.disabled,
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true),
+            choices = {
+              @TemplateProperty.DropdownPropertyChoice(
+                  value = "TEXTRACT_DETECT_DOCUMENT_TEXT",
+                  label = "Detect document text"),
+              @TemplateProperty.DropdownPropertyChoice(
+                  value = "TEXTRACT_ANALYZE_DOCUMENT",
+                  label = "Analyze document"),
+              @TemplateProperty.DropdownPropertyChoice(value = "NO_DATA", label = "")
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java:36]: Yes, this is a part of non mandatory field [DocumentReaderConfig]([DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax)), so it's fair to give users the option to leave it blank

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+            feel = Property.FeelMode.disabled,
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true),
+            choices = {
+              @TemplateProperty.DropdownPropertyChoice(
+                  value = "SERVICE_DEFAULT",
+                  label = "Service default"),
+              @TemplateProperty.DropdownPropertyChoice(
+                  value = "FORCE_DOCUMENT_READ_ACTION",
+                  label = "Force document read action"),
+              @TemplateProperty.DropdownPropertyChoice(value = "NO_DATA", label = ""),
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java:57]: Yes, this is a part of non mandatory field [DocumentReaderConfig]([DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax)), so it's fair to give users the option to leave it blank

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (johnBgood) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java:36]: I think it's more a UX question, maybe @YanaSegal has an opinion: should the dropdown value by empty, or something like "None" ?

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+    implements ComprehendCaller<ClassifyDocumentResult, ComprehendSyncRequestData> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SyncComprehendCaller.class);
+
+  @Override
+  public ClassifyDocumentResult call(
+      AmazonComprehendClient client, ComprehendSyncRequestData requestData) {
+    LOGGER.debug(
+        "Starting sync comprehend task for document classification with request data: {}",
+        requestData);
```
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: It seems to me that confidential data is passwords or tokens and the like. This ComprehendSyncRequestData does not contain such. From my point of view, the more data we have, the easier it is for us to support the product. An example with `text`, let's imagine that the user has passed text for processing that [DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax)  properties that he did not pass. Having this information, we can give him advice.

File: connectors/aws/aws-comprehend/README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/README.md]: amnesty, it's already been done https://github.com/camunda/connectors/blob/96d31e51167b2c5ff9c1d8002e281a7368d18eed/connectors/aws/aws-comprehend/README.md?plain=1#L3

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (johnBgood) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: In the context of [GDPR](https://gdpr-info.eu/issues/personal-data/) for instance (for Europe), Personal data can be anything like emails, names, date of birth, phone number, etc.

I agree that the more data we have the better (this is why I suggested overriding `toString` to log some information still), but we can't ignore the law.

[the remark this answers is not in the document]
Comment (johnBgood) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: > let's imagine that the user has passed text for processing that [DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax) properties that he did not pass. Having this information, we can give him advice.

In this case, we should rely on proper validation, not on logs.

[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: > > let's imagine that the user has passed text for processing that [DocumentReaderConfig](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_ClassifyDocument.html#API_ClassifyDocument_RequestSyntax) properties that he did not pass. Having this information, we can give him advice.
> 
> In this case, we should rely on proper validation, not on logs.

I don't think we should do this, it's too complicated and unexpected

[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: > In the context of [GDPR](https://gdpr-info.eu/issues/personal-data/) for instance (for Europe), Personal data can be anything like emails, names, date of birth, phone number, etc.
> 
> I agree that the more data we have the better (this is why I suggested overriding `toString` to log some information still), but we can't ignore the law.

Thanks for the documentation, I will override `toString` ))

[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/SyncComprehendCaller.java:26]: done

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (YanaSegal) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java:36]: Yes, please label the field even if it is optional. if nothing is selected we usually show a hint to the CTA - something like "Please select an option". as I don't have the context of the dropdown I can't be more specific. "None" works too.

[the remark this answers is not in the document]
Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendAsyncRequestData.java:36]: so the label "None" would be appropriate

File: connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java (hunk 16 above)
Comment (sbuettner) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java:61]: @DenovVasil I meant that `prepareDocumentReaderConfig` allows `null`. Then why the need to wrap it?

Comment (DenovVasil) [reply connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/caller/ComprehendCaller.java]: done

**Review by sbuettner — DISMISSED** [review]

**Review by johnBgood — APPROVED** [review]

**Pull request merged**